2-mandatory/4-stocks.js

```diff
@@ -65,6 +86,27 @@ function getPriceChanges(closingPricesForAllStocks) {
 */
 function highestPriceDescriptions(closingPricesForAllStocks, stocks) {
     // TODO
+    let theHighestPrice = highestPrice(closingPricesForAllStocks);
+    console.log(theHighestPrice)
+    let report = [];
+    let i = 0;
+    for (let stock of stocks) {
+        report.push("The highest price of " + stock.toUpperCase() + " in the last 5 days was " + theHighestPrice[i])
+        i++;
+    }
+    return report;
+
+}
+
+function highestPrice(closingPrices) {
+    highestPrices = [];
+    for (let price of closingPrices) {
+        highestPrices.push(/*Number*/((Math.max.apply(null, price)).toFixed(2))) // Found this solution here: https://stackoverflow.com/questions/1669190/find-the-min-max-element-of-an-array-in-javascript
+        // It works but I don't understand it - If I kept the Number method it would show the last higestPrice with one decimal after the point; not two...
```

Comment:
When you have Number and you're using toFixed(2) method on Number. You will have 2 decimal number after dot only if they are not including zero at the end. 
Examples of numbers:
10.563 => 10.56
10.100 => 10.1

When you have String - you will have two decimalsafter dot in any case. But the number will be a String.
"10.563" => "10.56"
"10.100" => "10.10"

So that's why it is better to leave the highest price as a String. Because in test we have a price "1101.30", not "1101.3". And we should only to show the price in sentence, we don't have any test which is checking if the price is a number.